src/libfuncs/felt252.rs

```diff
@@ -218,21 +219,21 @@ pub fn build_binary_operation<'ctx, 'this>(
             // For the initial setup, r0 = PRIME, r1 = a
             // This order is chosen because if we reverse them, then the first iteration will just swap them
             let prev_remainder = start_block
-                .append_operation(arith::constant(context, attr_prime_i512, location))
+                .append_op_result(arith::constant(context, attr_prime_i512, location))
```

Comment:
When you use append_op_result it removes the need to use `.result(0)?.into()` which is the point of this method. Please fix this on all the other places too, thanks!